### ISSUE TITLE
How do you parse out all matching substrings (not a full string match) with Lark?

### ISSUE THREAD

**Issue by enjoysmath** [lark-parser/lark#483]
Hi,

I'm parsing out variables from LaTeX (mathjax) expressions:

```
calc_grammar = """
    ?start: name
    ?name: (NAME | greek)   -> variable
    ?greek: GREEK_LOWER     -> greek
          | GREEK_UPPER     -> greek
    GREEK_LOWER: /\\\\alpha|\\\\beta|\\\\gamma|\\\\delta|\\\\epsilon|\\\\zeta|\\\\eta|\\\\theta/
               | /\\\\iota|\\\\kappa|\\\\lambda|\\\\mu|\\\\xi|\\\\omicron|\\\\pi|\\\\rho|\\\\sigma/
               | /\\\\tau|\\\\upsilon|\\\\phi|\\\\psi|\\\\chi|\\\\omega/
    GREEK_UPPER: /\\\\Alpha|\\\\Beta|\\\\Gamma|\\\\Delta|\\\\Epsilon|\\\\Zeta|\\\\Eta|\\\\Theta/
               | /\\\\Iota|\\\\Kappa|\\\\Lambda|\\\\Mu|\\\\Xi|\\\\Omicron|\\\\Pi|\\\\Rho|\\\\Sigma/
               | /\\\\Tau|\\\\Upsilon|\\\\Phi|\\\\Psi|\\\\Chi|\\\\Omega/
    ZZ: /\\\\Bbb{Z}/
    %import common.CNAME -> NAME
    %import common.WS_INLINE
    %ignore WS_INLINE
"""
```
But in order to work on general expressions, and not get a Lark grammar error, it seems like you have to add all the operators like *, +, - ie for every possible expression that might occur.  That is definitely NOT what I'd like to do.  I just want to grab all variables and leave the remaining text alone.  However, the obvious way of doing that, fails with Lark grammar errrors.

Any way around this?

**Comment by erezsh** [comment]
> it seems like you have to add all the operators like *, +, - ie for every possible expression that might occur

Absolutely not. What makes you think that?

**Comment by enjoysmath** [comment]
> > it seems like you have to add all the operators like *, +, - ie for every possible expression that might occur
> 
> Absolutely not. What makes you think that?

Could you post a simple example using "[a-z]" as a variable match for simplicity?  I have tried this before.  Please guide me! :)

**Comment by erezsh** [comment]
I'm not sure what you mean.

Have you taken a good look at the calc example? https://github.com/lark-parser/lark/blob/master/examples/calc.py

Why isn't it sufficient for your needs?

**Comment by enjoysmath** [comment]
@erezsh that happens just to be the example I expanded upon, however most of it goes away.

All I want to do is parse out "certain expressions" involving variables such as "a : \Bbb{Z}", treating all surrounding strings as constant and kept.  The end result I want is an array:

[Variable("a"), ":", ZZ()]

Of course this array can get much much longer.  However, it's important that I've parsed out the variables as these can then be substituted upon (in code) and different expressions recognized as being "the same as" a : \Bbb{Z}, such as "c : \Bbb{Z}".
